inc/saimirror.h

```diff
@@ -321,6 +326,28 @@ typedef enum _sai_mirror_session_attr_t
      */
     SAI_MIRROR_SESSION_ATTR_POLICER,
 
+    /**
+     * @brief UDP source port. UDP source port to encapsulate
+     * SFLOW packets
+     *
+     * @type sai_uint16_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     * @isvlan false
+     * @condition SAI_MIRROR_SESSION_ATTR_TYPE == SAI_MIRROR_SESSION_TYPE_ENHANCED_REMOTE
+     */
+    SAI_MIRROR_SESSION_ATTR_UDP_SRC_PORT,
```

Comment:
In most IP-UDP tunnels, a hash calculated over the original packet fields can be used to populate the UDP source port for better load-balancing. Do we want a similar capability here?

Reply:
@ashutosh-agrawal. Sure. I can update this proposal or add a new one. What do you suggest ?

Shall I go ahead and add another attribute SAI_MIRROR_SESSION_ATTR_UDP_SRC_PORT_ENTROPY_ENABLE ? or setting SAI_MIRROR_SESSION_ATTR_UDP_SRC_PORT to 0 will enable entropy hashing ?

Reply:
A new attribute is not necessary. You can just document the expected behavior of the device when SRC_PORT is set to 0.

Reply:
this create backward compatibility issue since now existing application must to supply this attribute now, otherwise it will fail?

Reply:
also for existing attribute SAI_MIRROR_SESSION_ATTR_GRE_PROTOCOL_TYPE since it is not needed for SFLOW_TUNNEL type, we should add condition there as well.

Reply:
@lguohan Add the conditional check now. UDP src port/dst port is needed only for sflow tunnel and gre protocol is needed only for l3 gre tunnel.